File: XMAN/re/reverse_box/list.py

def ror(num, size):
    num = num & 0xff
    tp = int('1'*size, 2)
    tp = num & tp
    num = num >> size
    tp = tp << (8-size)
    out = num | tp
    return out


for num in range(0xd9, 0xd9+1):
    a1 = [0 for i in range(256)]
    a1[0] = num
    idx = 1
    v8 = 1
    while True:
        v2 = (idx ^ 2 * idx) & 0xff
        if ((idx & 0x80) == 0):
            v3 = 0
        else:
            v3 = 27
        idx = (v2 ^ v3) & 0xff
        v4 = (4 * (2 * v8 ^ v8) ^ 2 * v8 ^ v8) & 0xff
        v9 = (16 * v4 ^ v4) & 0xff
        if (v9 >= 0):
            v5 = 0
        else:
            v5 = 9
        v8 = v9 ^ v5
        result = ror(v8, 4) ^ ror(v8, 5) ^ ror(v8, 6) ^ ror(v8, 7) ^ (v8 ^ a1[0])
        a1[idx] = result
        if idx == 1:
            break
    # print(hex(a1[ord('T')]))
    # print(hex(a1[ord('W')]))
    # print(hex(a1[ord('C')]))
    # print(hex(a1[ord('T')]))
    # print(hex(a1[ord('F')]))
    print(a1)


'''
95eeaf95ef94234999582f722f492f72b19a7aaf72e6e776b57aee722fe77ab5ad9aaeb156729676ae7a236d99b1df4a
'''

File: XMAN/re/reverse_box/list.c
#include <stdio.h>
int ror(int num, int size)
{
    char tp = 0;
    for (int i = 0; i < size; i++)
    {
        tp += 1 << i;
    }
    tp = num & tp;
    num = num >> size;
    tp = tp << (8 - size);
    return num | tp;
}
int main()
{
    unsigned int v1;    // eax
    int v2;             // edx
    char v3;            // al
    char v4;            // ST1B_1
    char v5;            // al
    int result;         // eax
    unsigned __int8 v7; // [esp+1Ah] [ebp-Eh]
    char v8;            // [esp+1Bh] [ebp-Dh]
    char v9;            // [esp+1Bh] [ebp-Dh]
    int v10;            // [esp+1Ch] [ebp-Ch]

    int bp = 156;
    for (int bp = 0xd9; bp <= 0xd9; bp++)
    {
        // printf("%d\n",bp);
        int list[256] = {0};
        list[0] = bp;

        char bVar1;
        unsigned __seed;
        char local_12;
        char local_11;
        local_12 = 1;
        local_11 = 1;
        do
        {
            if ((char)local_12 < '\0')
            {
                bVar1 = 0x1b;
            }
            else
            {
                bVar1 = 0;
            }
            local_12 = bVar1 ^ local_12 * '\x02' ^ local_12;
            bVar1 = local_11 ^ local_11 * '\x02';
            bVar1 = bVar1 ^ bVar1 << 2;
            bVar1 = bVar1 ^ bVar1 << 4;
            if (bVar1 < '\0')
            {
                local_11 = 9;
            }
            else
            {
                local_11 = 0;
            }
            local_11 = local_11 ^ bVar1;
            list[local_12] = (list[0] ^ local_11 ^ (bVar1 >> 7 | local_11 << 1) ^ (bVar1 >> 6 | local_11 << 2) ^ (bVar1 >> 5 | local_11 << 3) ^ (bVar1 >> 4 | local_11 << 4)) & 0xff;
        } while (local_12 != 1);
        for(int i=0;i<256;i++)
        {
            printf("%d,",list[i]);
        }
    }
    return result;
}
